### PR TITLE
[PVR] Trac #17374: enabled "PVR -> Power saving -> Wakeup command" setting is not called

### PR DESCRIPTION
**Pull request by ksooo**
This should finally fix http://trac.kodi.tv/ticket/17374. First fix attempt (#11849) was incomplete.

This is a more complete approach, which introduces a self contained auto updating settings container for PVR settings.

@FernetMenta for review, please.

### CODE REVIEW

**Comment by FernetMenta** [comment]
@ksooo I will have a look soon

**Comment by ksooo** [comment]
@FernetMenta i have some ideas to dramatically simplify this. no need for a review atm. will ping you when I think I'm done.

**Comment by ksooo** [comment]
Added some more commits:

1. get rid of CPVRGUIActions static singleton. Caused crashes on exit while destructing CPVRGUIAction's new CPVRSettings member while trying to deregister listener from already destructed settings manager

2. Factor out a minimal timers container and use it instead of CPVRTimers when updating timers, Saves resources and performs better (for example, no need to register as observer and setting listener etc while doing timers update).

3. Fix a memeory leak in CPVRTimers' timer tag map.

**Comment by ksooo** [comment]
4. Fix destruction of windows on app exit (windows were queued for deletion, but dtors were never called, because async destruction does not work anymore on app exit).

**Comment by ksooo** [comment]
@FernetMenta before this PR gets to big (stumbling over bugs and glitches again and again while working on the root cause of trac17374 ...) I want to hear your opinion.

File: xbmc/pvr/PVRGUIActions.cpp [hunk 1]
```diff
@@ -450,9 +453,9 @@ namespace PVR
       if (bOnOff && !channel->IsRecording())
       {
         CEpgInfoTagPtr epgTag;
-        int iDuration = CServiceBroker::GetSettings().GetInt(CSettings::SETTING_PVRRECORD_INSTANTRECORDTIME);
+        int iDuration = m_settings.GetIntValue(CSettings::SETTING_PVRRECORD_INSTANTRECORDTIME);
```
Comment (xhaggi) [inline xbmc/pvr/PVRGUIActions.cpp]: Why do you introduce this and all the stuff in the PVRSettings. Why do we need a container for the PVR related settings? If you want to get the current setting value, all you need to do this calling this or the other type related setting methods. What do you want to achieve with it?

Comment (ksooo) [inline xbmc/pvr/PVRGUIActions.cpp]: Well, please read #11849 for the explanation.

Comment (xhaggi) [inline xbmc/pvr/PVRGUIActions.cpp]: Thx will do

**Comment by ksooo** [comment]
I removed the not directly related commits from this PR and made separate PRs out of them:

- #11918
- #11919 
- #11920

**Comment by FernetMenta** [comment]
@ksooo I only had time for a brief look at it. I think your changes are fine. Nice that you dropped another singleton that makes PVR a more self contained component.

**Comment by ksooo** [comment]
rebased

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
Ay objctions? Otherwise I will merge this the next days.

**Comment by ksooo** [comment]
jenkins build this please

**Pull request merged**